[Spaces and Roles] Fix rules on showing `Permissions` tab

tsullivan opened the pull request:
## Summary

Some deployment types don't have custom roles. This PR hides the `Permissions` tab in Spaces Management for projects without role management enabled. The solution uses the isRoleManagementEnabled function from the security plugin’s authz service for this.


<!--ONMERGE {"backportTargets":["8.16","8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242872)
* Commit: 5b58f75adc02354079dee6553a5d20b903889470

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/242872#019291f3-8906-4c2a-9640-b936012bcf0e)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":242872}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242898)
* Commit: 783a9734dc1a6131c13d0976d87cb5de7e5a5723

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/242898#0192925a-db8a-4182-b822-46dbc06c6730)

### History
* :broken_heart: [Build #242872](https://buildkite.com/elastic/kibana-pull-request/builds/242872) failed 5b58f75adc02354079dee6553a5d20b903889470

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242898","number":242898,"commit":"783a9734dc1a6131c13d0976d87cb5de7e5a5723"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":242898}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242914)
* Commit: a35c0d3f1c32cddee27ce00a7422d68b27eb6f95

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/242914#019292af-d1b9-413c-98a0-8837aca62269)

### History
* :broken_heart: [Build #242898](https://buildkite.com/elastic/kibana-pull-request/builds/242898) failed 783a9734dc1a6131c13d0976d87cb5de7e5a5723
* :broken_heart: [Build #242872](https://buildkite.com/elastic/kibana-pull-request/builds/242872) failed 5b58f75adc02354079dee6553a5d20b903889470

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242914","number":242914,"commit":"a35c0d3f1c32cddee27ce00a7422d68b27eb6f95"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242898","number":242898,"commit":"783a9734dc1a6131c13d0976d87cb5de7e5a5723"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":242914}
buildkite-pr-comment-->

jeramysoucy commented on x-pack/plugins/spaces/public/management/edit_space/edit_space_tabs.tsx:
```diff
@@ -92,7 +93,11 @@ export const getTabs = ({
 
-  if (canUserViewRoles) {
+  const canUserViewRoles = Boolean(capabilities?.roles?.view);
+  const isRoleManagementEnabled = authz && authz.isRoleManagementEnabled() !== false;
```
```suggestion
  const isRoleManagementEnabled = authz && authz.isRoleManagementEnabled();
```

tsullivan replied:
@jeramysoucy I think we examined this further in Slack, and agreed that the `config.roleManagementEnabled` is only available in serverless, which makes the `authz.isRoleManagementEnabled()` return undefined in stateful, when it actually should be interpreted as being enabled.

So how about:
```
  /*
   * config.roleManagementEnabled is an offering-based config. In serverless, the value will be true or false.
   * In stateful, role management is enabled but the config setting is expected to be undefined.
   */
  const isRoleManagementEnabled =
    authz &&
    (typeof authz.isRoleManagementEnabled() === 'undefined' || authz.isRoleManagementEnabled());
```

Alternatively, we could update the implementation of `authz.isRoleManagementEnabled` in https://github.com/elastic/kibana/blob/4339f8465720c507351a2e65a6fe881a10dcd938/x-pack/plugins/security/public/authorization/authorization_service.ts#L21

tsullivan replied:
Moved this logic to x-pack/plugins/spaces/public/management/edit_space/provider/edit_space_provider.tsx

jeramysoucy replied:
Thanks, Tim! Sorry for the confusion.

eokoneyo commented on x-pack/plugins/spaces/public/management/spaces_management_app.tsx:
```diff
@@ -73,6 +74,10 @@ export const spacesManagementApp = Object.freeze({
           import('./edit_space'),
         ]);
 
+        const { security } = await coreStart.plugins.onStart<{ security: SecurityPluginStart }>(
+          'security'
+        );
+
```
I think we should stick to the same pattern we've adopted to inject utils exposed by the security plugin, through the space plugin entry [x-pack/plugins/spaces/public/plugin.tsx](https://github.com/elastic/kibana/blob/main/x-pack/plugins/spaces/public/plugin.tsx#L93-L115), wdyt?

tsullivan replied:
Good suggestion, it is much cleaner now.

eokoneyo commented on x-pack/plugins/spaces/tsconfig.json:
```diff
@@ -50,7 +50,8 @@
     "@kbn/logging",
     "@kbn/core-logging-browser-mocks",
-    "@kbn/core-http-router-server-mocks"
+    "@kbn/core-http-router-server-mocks",
+    "@kbn/security-plugin"
```
This might trigger a cyclic dependency warning, since the security plugin also depends on the spaces plugin

eokoneyo reviewed (commented):
Left some nit comments

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243213)
* Commit: db8c2f94382db7266df5d6b310b322977d91ae96

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/243213#01929637-3722-4e9e-9a17-a13d8fc838ea)

### History
* :broken_heart: [Build #242914](https://buildkite.com/elastic/kibana-pull-request/builds/242914) failed a35c0d3f1c32cddee27ce00a7422d68b27eb6f95
* :broken_heart: [Build #242898](https://buildkite.com/elastic/kibana-pull-request/builds/242898) failed 783a9734dc1a6131c13d0976d87cb5de7e5a5723
* :broken_heart: [Build #242872](https://buildkite.com/elastic/kibana-pull-request/builds/242872) failed 5b58f75adc02354079dee6553a5d20b903889470

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243213","number":243213,"commit":"db8c2f94382db7266df5d6b310b322977d91ae96"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242914","number":242914,"commit":"a35c0d3f1c32cddee27ce00a7422d68b27eb6f95"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242898","number":242898,"commit":"783a9734dc1a6131c13d0976d87cb5de7e5a5723"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":243213}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/kibana-security (Team:Security)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243279)
* Commit: eeb43a891394aeff70d2b76aa9cf8908b2d84c5c

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/456e9b81c2f6ed7bb880fadef342a3ee1cadde6d) | [after](https://github.com/elastic/kibana/commit/eeb43a891394aeff70d2b76aa9cf8908b2d84c5c) | diff |
| --- | --- | --- | --- |
| `spaces` | 256.2KB | 256.4KB | +215.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/456e9b81c2f6ed7bb880fadef342a3ee1cadde6d) | [after](https://github.com/elastic/kibana/commit/eeb43a891394aeff70d2b76aa9cf8908b2d84c5c) | diff |
| --- | --- | --- | --- |
| `spaces` | 33.7KB | 34.1KB | +347.0B |



### History
* :broken_heart: [Build #243213](https://buildkite.com/elastic/kibana-pull-request/builds/243213) failed db8c2f94382db7266df5d6b310b322977d91ae96
* :broken_heart: [Build #242914](https://buildkite.com/elastic/kibana-pull-request/builds/242914) failed a35c0d3f1c32cddee27ce00a7422d68b27eb6f95
* :broken_heart: [Build #242898](https://buildkite.com/elastic/kibana-pull-request/builds/242898) failed 783a9734dc1a6131c13d0976d87cb5de7e5a5723
* :broken_heart: [Build #242872](https://buildkite.com/elastic/kibana-pull-request/builds/242872) failed 5b58f75adc02354079dee6553a5d20b903889470

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243279","number":243279,"commit":"eeb43a891394aeff70d2b76aa9cf8908b2d84c5c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243213","number":243213,"commit":"db8c2f94382db7266df5d6b310b322977d91ae96"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242914","number":242914,"commit":"a35c0d3f1c32cddee27ce00a7422d68b27eb6f95"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242898","number":242898,"commit":"783a9734dc1a6131c13d0976d87cb5de7e5a5723"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":243279}
buildkite-pr-comment-->

jeramysoucy commented on x-pack/plugins/spaces/public/management/edit_space/edit_space_roles_tab.test.tsx:
Nit: we could a test here to confirm that the `Permissions` tab does not render when `getIsRoleManagementEnabled` resolves `false`.

jeramysoucy reviewed (approved):
LGTM. Just one nit/suggestion to add a unit test case.

tsullivan commented:
@jeramysoucy I have addressed the feedback by adding tests to unit tests that deal with the permissions tab.

I have also updated the code because the original fix didn't account for an HTTP request that results in a 404 when role management is disabled: 3795630cf82

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243660)
* Commit: 9efeb5909ac6b3c5e868377024080ccd24a3befb

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1a1cba7b565b3028220d3cc6ae009554d00ee28f) | [after](https://github.com/elastic/kibana/commit/9efeb5909ac6b3c5e868377024080ccd24a3befb) | diff |
| --- | --- | --- | --- |
| `spaces` | 256.2KB | 256.4KB | +234.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/1a1cba7b565b3028220d3cc6ae009554d00ee28f) | [after](https://github.com/elastic/kibana/commit/9efeb5909ac6b3c5e868377024080ccd24a3befb) | diff |
| --- | --- | --- | --- |
| `spaces` | 33.7KB | 34.1KB | +347.0B |



### History
* :green_heart: [Build #243279](https://buildkite.com/elastic/kibana-pull-request/builds/243279) succeeded eeb43a891394aeff70d2b76aa9cf8908b2d84c5c
* :broken_heart: [Build #243213](https://buildkite.com/elastic/kibana-pull-request/builds/243213) failed db8c2f94382db7266df5d6b310b322977d91ae96
* :broken_heart: [Build #242914](https://buildkite.com/elastic/kibana-pull-request/builds/242914) failed a35c0d3f1c32cddee27ce00a7422d68b27eb6f95
* :broken_heart: [Build #242898](https://buildkite.com/elastic/kibana-pull-request/builds/242898) failed 783a9734dc1a6131c13d0976d87cb5de7e5a5723
* :broken_heart: [Build #242872](https://buildkite.com/elastic/kibana-pull-request/builds/242872) failed 5b58f75adc02354079dee6553a5d20b903889470

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243660","number":243660,"commit":"9efeb5909ac6b3c5e868377024080ccd24a3befb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243279","number":243279,"commit":"eeb43a891394aeff70d2b76aa9cf8908b2d84c5c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243213","number":243213,"commit":"db8c2f94382db7266df5d6b310b322977d91ae96"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242914","number":242914,"commit":"a35c0d3f1c32cddee27ce00a7422d68b27eb6f95"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242898","number":242898,"commit":"783a9734dc1a6131c13d0976d87cb5de7e5a5723"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242872","number":242872,"commit":"5b58f75adc02354079dee6553a5d20b903889470"}],"number":243660}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.16, 8.x

https://github.com/elastic/kibana/actions/runs/11406746377

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.16|Backport failed because of merge conflicts|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 196442
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

tsullivan commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/196937">](https://github.com/elastic/kibana/pull/196937)|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/196938">](https://github.com/elastic/kibana/pull/196938)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
